src/main/java/org/opensearch/knn/index/codec/KNN80Codec/KNN80DocValuesConsumer.java

```diff
@@ -243,18 +242,6 @@ private void createKNNIndexFromScratch(FieldInfo fieldInfo, KNNCodecUtil.Pair pa
             );
         }
 
-        // Update index description of Faiss for binary data type
-        if (KNNEngine.FAISS == knnEngine
```

Comment:
what will happen if segment is an old segment. I think we will still need this update for old segments. We might need to find a way to check if the index description is correct for binary index and if it not correct then update it otherwise don't update the index description.